django/db/models/fields/related_descriptors.py

```diff
@@ -311,7 +317,7 @@ def RelatedObjectDoesNotExist(self):
         )
 
     def is_cached(self, instance):
-        return hasattr(instance, self.cache_name)
+        return self.related.is_cached(instance)
```

Comment:
Shouldn't this be `self.related.field.is_cached(instance)`?

And if so, does that reveal a lack of test coverage here that should be fixed?